StripePayments/StripePayments/Source/Helpers/STPCardValidator.swift

```diff
@@ -414,6 +414,43 @@ public class STPCardValidator: NSObject {
         return Set(brands)
     }
 
+    // This is a bit of a hack: We want to fetch BIN information for Card Brand Choice, but some
+    // of the BIN length information coming from the service is incorrect: We're receiving a maximum
+    // length, but we really should receive a min-max range.
+    // We don't want to pollute the main STPBINController cache with this bad data.
+    //
+    // We currently prevent cache pollution with an `isVariableLengthBINPrefix` check in
+    // `retrieveBinRanges()`, but we'll bypass that check when using the CBC BIN controller.
+    static let cbcBinController = STPBINController()
+
+    /// Returns available brands for the provided card details.
+    /// - Parameter card: The card details to validate.
+    /// - Parameter completion: Will be called with the set of available STPCardBrands or an error.
+    /// - seealso: https://stripe.com/docs/card-brand-choice
+    public class func possibleBrands(forCard cardParams: STPPaymentMethodCardParams,
+                                     completion: @escaping (Result<Set<STPCardBrand>, Error>) -> Void) {
+        guard let cardNumber = cardParams.number else {
+            completion(.success([]))
```

Comment:
Is it a success if no number is passed in?

Reply:
Good question: If you pass an empty string, the answer is `.allCases`, as any card brand is possible. A nil string should probably match that behavior. The existing `brandForNumber` API returns `.unknown` if multiple brands match.

It'd be messy ergonomically to make this an error: The user isn't creating the `STPPaymentMethodCardParams` on their own, they're receiving it from one of our UI fields, so an empty card field would return an empty/nil string. The user would then need to add a special case to access `cardParams.number` (which we discourage) to check for a nil or empty string before calling `possibleBrands()`, or they would need to handle "empty card number" errors for any empty card field, which they'd probably want to treat separately from network errors.

Reply:
I feel like a comment here is probably worth including

Reply:
I added one before merge but lmk if you think it's unclear, I can always go back and add more context: https://github.com/stripe/stripe-ios/pull/2373/files#diff-974d2c2ad6918ec7bf57dd1c8bc19f370013acf59c601b8b7c6941be2fd5cfbeR433

Reply:
Oh nice! Didn't realize you added already